libs/blocks/marquee/marquee.js

```diff
@@ -137,8 +137,8 @@ export default function init(el) {
     media.classList.add('media');
     const video = media.querySelector('video');
     if (video) applyHoverPlay(video);
-    if (media.querySelector('a[href$=".mp4"]')) {
-      decorateVideo(media);
+    if (media.querySelector('a[href*=".mp4"]')) {
+      decorateVideo(media, media.querySelector('a').href);
```

Comment:
hoverplay and autoplay1 is not working intermittently in marquee block. This appears to be because of the race situation for video autoblock in a block we discussed yesterday. With this change if video.js is not called first video block is not created. Hence hoverplay event is not added at line 139. However at line 141 we are creating video tag in marquee.js itself for mp4 videos with hash value and video.js is not called and events not added. In marquee.js (decorateVideo function) we are hardcoding video block with autoplay on loop. Hence any hash value other than autoplay is not working as expected

Test page to see the behavior: https://videofix--milo--adobecom.hlx.page/drafts/ruchika/marquee1

Reply:
fixed